botocove/cove_host_account.py

```diff
@@ -60,7 +48,16 @@ def __init__(
             self.target_regions = regions
 
         self.provided_ignore_ids = ignore_ids
-        self.target_accounts = self._resolve_target_accounts(target_ids)
+        self.provided_target_ids = target_ids
+
+        if not org_master and self.provided_target_ids:
```

Comment:
It turns out that some of the tests do use `org_master=False` but not how you might expect.

The tests don't pass any `target_ids` so botocove ends up listing the organization accounts anyway!

That seems silly and would surely fail in a non-org host account with an error along the lines of "org service not in use".

For now I want to retain that behavior, so that's why this strange if statement is here.

At least the weirdness is a  more explicit now :-)

Reply:
org_master definitely has a vestigial feel to it: with this change, I think we can deprecate it and just log a warning if it's set to False as a no-op :)

The readme does sort of cover why it's there today though and the contradiction you're seeing- essentially to cop out of the problem you were having with rate limiting, but from a test point of view it's just easier to not have to mock out the describe account call... failing with an IAM error in that sense is sort of reasonable at least :)

> org_master: bool
> 
> Defaults to True. When True, will leverage the Boto3 Organizations API to list all accounts in the organization, and enrich each CoveSession with information available (Id, Arn, Name, Status, Email). Disabling this and providing your own full list of accounts may be a desirable optimisation if speed is an issue.
> 
> org_master=False means only Id will be available to CoveSession.`